contracts/test/TestJBveBanny.t.sol

```diff
@@ -47,137 +61,1022 @@ contract JBveBannyTests is TestBaseWorkflow {
-    (, uint256 d, uint256 lockedUntil,,) = _jbveBanny.getSpecs(1);
-    assertEq(d, 864000);
-    evm.warp(lockedUntil / 2);
-    evm.expectRevert(abi.encodeWithSignature('INVALID_LOCK_EXTENSION()'));
+    mintAndApproveIJBTokens();
+    vm.startPrank(_projectOwner);
+    _jbveBanny.lock(_projectOwner, 10 ether, 1 weeks, _projectOwner, true, false);
+    (, uint256 _duration, uint256 _lockedUntil, , ) = _jbveBanny.getSpecs(1);
+    assertEq(_duration, 1 weeks);
+    vm.warp(_lockedUntil / 2);
+    vm.expectRevert(abi.encodeWithSignature('INVALID_LOCK_EXTENSION()'));
 
     JBLockExtensionData[] memory extends = new JBLockExtensionData[](1);
-    extends[0] = JBLockExtensionData(
-      1, 8640000
-    );
+    extends[0] = JBLockExtensionData(1, 4 weeks);
     _jbveBanny.extendLock(extends);
+    vm.stopPrank();
   }
 
+  /**
+    @dev Lock with non-Jbtoken Test.
+  */
   function testLockWithNonJbToken() public {
     _projectOwner = projectOwner();
-    evm.startPrank(_projectOwner);
+    vm.startPrank(_projectOwner);
     _jbController.mintTokensOf(_projectId, 100 ether, _projectOwner, 'Test Memo', false, true);
     uint256[] memory _permissionIndexes = new uint256[](1);
     _permissionIndexes[0] = JBOperations.TRANSFER;
     jbOperatorStore().setOperator(
       JBOperatorData(address(_jbveBanny), _projectId, _permissionIndexes)
     );
-    _jbveBanny.lock(_projectOwner, 10 ether, 864000, _projectOwner, false, false);
+    _jbveBanny.lock(_projectOwner, 10 ether, 1 weeks, _projectOwner, false, false);
+    assertGt(_jbveBanny.tokenVotingPowerAt(1, block.number), 0);
+    (int128 _amount, , uint256 _duration, bool _useJbToken, bool _allowPublicExtension) = _jbveBanny
+      .locked(1);
+    assertEq(_amount, 10 ether);
+    assertEq(_duration, 1 weeks);
+    assertTrue(!_useJbToken);
+    assertTrue(!_allowPublicExtension);
     assertEq(_jbveBanny.ownerOf(1), _projectOwner);
-    (uint256 amount, uint256 duration, , ,) = _jbveBanny.getSpecs(1);
-    assertEq(amount, 10 ether);
-    assertEq(duration, 864000);
+    (uint256 _specsAmount, uint256 _specsDuration, , , ) = _jbveBanny.getSpecs(1);
+    assertEq(_specsAmount, 10 ether);
+    assertEq(_specsDuration, 1 weeks);
+    vm.stopPrank();
+  }
+
+  /**
+    @dev Voting power increase Test.
+  */
+  function testLockVotingPowerIncreasesIfLockedLonger() public {
+    mintAndApproveIJBTokens();
+
+    vm.startPrank(_projectOwner);
+
+    uint256 _tokenId = _jbveBanny.lock(
+      _projectOwner,
+      5 ether,
+      1 weeks,
+      _projectOwner,
+      true,
+      false
+    );
+
+    // The unlock date gets rounded to the week, so it may not be exact
+    uint256 unlockDate = ((block.timestamp + 1 weeks) / 1 weeks) * 1 weeks;
+    // The exact amount of seconds until the 1 week lock expires
+    uint256 singleWeekLockExactSeconds = (unlockDate - block.timestamp);
+    uint256 expectedVotingPower = (5 ether / MAXTIME) * singleWeekLockExactSeconds;
```

Comment:
nit> reduce floating point precision error with other order of operations: `5 ether * singleWeekLockExactSeconds / MAXTIME`

Reply:
@drgorillamd I checked this and with this, there was a slight precision difference I checked the curve tests just to verify and they too used the same order we have so all good IMO